airflow/providers/amazon/aws/operators/ec2.py

```diff
@@ -254,3 +256,100 @@ def execute(self, context: Context):
                         "MaxAttempts": self.max_attempts,
                     },
                 )
+
+class EC2RebootInstanceOperator(BaseOperator):
+    """
+    Reboot AWS EC2 instance using boto3.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:EC2RebootInstanceOperator`
+
+    :param instance_id: id of the AWS EC2 instance
+    :param aws_conn_id: aws connection to use
+    :param region_name: (optional) aws region name associated with the client
```

Comment:
Why bother marking just this one as optional but not the arg above and below? I'd either do it for all optional args or don't bother. Same for Hibernate operator below

Reply:
I naively copied the `EC2StartInstanceOperator` setup. Switched to use the pattern established by more recent operators so there are no more optional args.